localtime/data_source_timestamp.go

```diff
@@ -68,10 +68,39 @@ func dataSourceTimestamp() *schema.Resource {
 }
 
 func dataSourceTimestampRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	// Warning or errors can be collected in a slice type
 	var diags diag.Diagnostics
 
-	// Add implementation details here
+	local_time := d.Get("local_time").(string)
+
+	layout := d.Get("layout").(string)
+	if layout == "" {
+		layout_local, err := dateparse.ParseFormat(local_time)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		layout = fmt.Sprintf("%s %s", layout_local, d.Get("layout_timezone"))
+	}
+
+	var location *time.Location
+
+	location_str := d.Get("location").(string)
+	if location_str == "" {
+		location = time.Local
+	} else {
+		location, _ = time.LoadLocation(location_str)
+	}
```

Comment:
```suggestion
	location := time.Local
	if v, ok := d.GetOk("location"); ok {
		location, _ = time.LoadLocation(v.(string))
	}
```

But also, don't ignore the error! You've validated early so it shouldn't occur, but it's easy to just return it in case.

Reply:
This is not strictly the same btw, it is not treating an empty string as a special value. `GetOk` will only return `_, true` when `location` is not null. An empty string should produce a validation error since it will cause `LoadLocation` to error.